packages/contracts-rfq/contracts/FastBridgeV2.sol

```diff
@@ -92,21 +112,28 @@ contract FastBridgeV2 is Admin, IFastBridgeV2, IFastBridgeV2Errors {
         if (protocolFeeRate > 0) originFeeAmount = (originAmount * protocolFeeRate) / FEE_BPS;
         originAmount -= originFeeAmount; // remove from amount used in request as not relevant for relayers
 
+        // Calculate exclusivity end time only if exclusivity params are set
+        uint256 exclusivityEndTime =
+            paramsV2.quoteExclusivitySeconds == 0 ? 0 : block.timestamp + paramsV2.quoteExclusivitySeconds;
         // set status to requested
         bytes memory request = abi.encode(
-            BridgeTransaction({
-                originChainId: uint32(block.chainid),
-                destChainId: params.dstChainId,
-                originSender: params.sender,
-                destRecipient: params.to,
-                originToken: params.originToken,
-                destToken: params.destToken,
-                originAmount: originAmount,
-                destAmount: params.destAmount,
-                originFeeAmount: originFeeAmount,
-                sendChainGas: params.sendChainGas,
-                deadline: params.deadline,
-                nonce: nonce++ // increment nonce on every bridge
+            BridgeTransactionV2({
+                txV1: BridgeTransaction({
+                    originChainId: uint32(block.chainid),
+                    destChainId: params.dstChainId,
+                    originSender: params.sender,
+                    destRecipient: params.to,
+                    originToken: params.originToken,
+                    destToken: params.destToken,
+                    originAmount: originAmount,
+                    destAmount: params.destAmount,
+                    originFeeAmount: originFeeAmount,
+                    sendChainGas: params.sendChainGas,
+                    deadline: params.deadline,
+                    nonce: nonce++ // increment nonce on every bridge
+                }),
+                exclusivityRelayer: paramsV2.quoteRelayer,
+                exclusivityEndTime: exclusivityEndTime
             })
```

Comment:
I initially took the nested approach because the explicit struct assignment was leading to "stack too deep" error. But now that I think about this, this could be reworked wothout nesting by unrolling the single assignment into a few separate ones like:
```solidity
    BridgeTransactionV2 memory txV2;
    txV2.originChainId = uint32(block.chainid);
    txV2.destChainId = params.dstChainId;
    // And so on
```

We do have coverage to ensure no assignments have been missed - and this might end up even being a bit less gas intensive (due to less stack operations).

Reply:
Will post an update here once I see if this ends up being more effective or not

Reply:
@parodime as it turns out, the explicit assignment wasn't the issue, but rather the auto-generated getter views for the test contracts (e.g. `IFastBridgeV2.BridgeTransactionV2 public tokenTx;`). Once these were converted into internal variables, the stack-too-deep error was gone. Refactored in 784457f6acf5f8eb283db8ac48fd91623ef78e67, the main contract diff is much cleaner now.